src/Action/DailyPostArchiveAction.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\NewsBundle\Action;
+
+use Sonata\IntlBundle\Templating\Helper\DateTimeHelper;
+use Sonata\NewsBundle\Model\BlogInterface;
+use Sonata\NewsBundle\Model\PostManagerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Translation\TranslatorInterface;
+
+final class DailyPostArchiveAction extends AbstractPostArchiveAction
+{
+    /**
+     * @var DateTimeHelper
+     */
+    private $dateTimeHelper;
+
+    public function __construct(
+        BlogInterface $blog,
+        PostManagerInterface $postManager,
+        TranslatorInterface $translator,
+        DateTimeHelper $dateTimeHelper
+    ) {
+        parent::__construct($blog, $postManager, $translator);
+
+        $this->dateTimeHelper = $dateTimeHelper;
+    }
+
+    /**
+     * @param string $year
+     * @param string $month
+     * @param string $day
+     *
+     * @return Response
+     */
+    public function __invoke(Request $request, $year, $month, $day)
+    {
+        $date = $this->getPostManager()->getPublicationDateQueryParts(sprintf('%d-%d-%d', $year, $month, $day), 'day');
+
+        if ($seoPage = $this->getSeoPage()) {
```

Comment:
Early return?

Reply:
Can you elaborate?

Reply:
Nevermind, it reviewed too quickly, an early return is not going to be better